tests/test_dle.py

```diff
@@ -731,9 +731,16 @@ def test_edge_cases(self, exprs, simd_level, expected):
         op = Operator(exprs, opt=('advanced', {'openmp': True}))
 
         iterations = FindNodes(Iteration).visit(op)
-        assert 'omp for collapse' in iterations[0].pragmas[0].value
-        if simd_level:
-            assert 'omp simd' in iterations[simd_level].pragmas[0].value
+        try:
+            assert 'omp for collapse' in iterations[0].pragmas[0].value
```

Comment:
Does that mean that you do not see "omp for collapse" in every case as before?
If this is the case I would prefer another variable in parametrization to whether "omp for collapse" is expected or not.
If we still see "omp for collapse" it should be out of the try-except.

Ideally we could use parameters to avoid the try-except. THese parameters would reflect the expected positions of omp and simd pragmas.

Reply:
false, it is caught in the except

Reply:
Then why do we not have:
```
        assert 'omp for collapse' in iterations[0].pragmas[0].value
        try:
            if simd_level:
                assert 'omp simd' in iterations[simd_level].pragmas[0].value
        except:
            # E.g. gcc-5 doesn't support array reductions, so the compiler will
            # generate different legal code
            assert not Ompizer._support_array_reduction(configuration['compiler'])
```

The "omp for collapse is always in iterations[0], right?

Reply:
If it just false we can add a similar var to simd_level, like omp_level to get either 0 or None...
i.e.
```
if omp_level:
    assert 'omp for collapse' in iterations[0].pragmas[0].value
```

Reply:
> The "omp for collapse is always in iterations[0], right?

no, with gcc-5 it will be in the innermost loop*, as it prefers thread parallelism over simd parallelism


* it's a bit more complicated than that... it actually depends on the number of physical and logical cores of the underlying platform, as collapsing may further change what the candidate-selector ends up choosing. Hence the loose(ish) `any(...)` in the except branch

Reply:
okk